.github/workflows/deploy_to_do.yml

```
@@ -1,159 +1,146 @@
-          oauth-secret: ${{ secrets.TS_OAUTH_SECRET }} # Tailscale OAuth secret
-          tags: tag:github-actions # Tag for this Tailscale node
-          version: latest # Use latest Tailscale version
-          use-cache: 'true' # Cache Tailscale for faster setup
-
-      - name: Debug ED25519 Key # Setup and verify SSH key
+          oauth-client-id: ${{ secrets.TS_OAUTH_CLIENT_ID }}  # OAuth client ID for Tailscale
+          oauth-secret: ${{ secrets.TS_OAUTH_SECRET }}  # OAuth client secret for Tailscale
+          tags: tag:github-actions  # Tag to identify this connection in Tailscale
+          version: latest  # Use the latest version of Tailscale
+          use-cache: 'true'  # Cache Tailscale binary for faster startup
+
+      # Step 2: (Debug only) Verifies SSH ED25519 SSH key
+      - name: Debug ED25519 Key  # SSH key debugging step
+        if: ${{ inputs.enable_ssh_debugging == true }}  # Only run when debugging is enabled
+        run: |
+          mkdir -p ~/.ssh  # Create SSH directory if it doesn't exist
+          echo "${{ secrets.DO_SSH_KEY }}" > ~/.ssh/id_ed25519  # Create private SSH key file
+          chmod 600 ~/.ssh/id_ed25519  # Set secure permissions on the key
+          echo "${{ secrets.DO_HOST_KEY }}" >> ~/.ssh/known_hosts  # Add host key to known hosts
+          ssh-keygen -l -f ~/.ssh/id_ed25519  # Show fingerprint of the key
+          ssh-keygen -y -f ~/.ssh/id_ed25519  # Show public key derived from private key
+          ssh -vvv -o StrictHostKeyChecking=accept-new -o BatchMode=yes -i ~/.ssh/id_ed25519 ${{ secrets.DO_USERNAME }}@${{ secrets.DO_TAILSCALE_NAME }} 'echo "Connection successful"'  # Test connection with verbose output
+
+      # Step 3: Create .env File Locally
+      - name: Create .env File Locally  # Create environment file for deployment
         run: |
-          mkdir -p ~/.ssh # Create SSH directory
-          echo "${{ secrets.DO_SSH_KEY }}" > ~/.ssh/id_ed25519 # Save SSH key
-          chmod 600 ~/.ssh/id_ed25519 # Set correct permissions
-          
-          echo "${{ secrets.DO_HOST_KEY }}" >> ~/.ssh/known_hosts # Add host key
-          
-          echo "Key information:" # Display key information for debugging
-          ssh-keygen -l -f ~/.ssh/id_ed25519 # Show key fingerprint
-          
-          echo "Public key that should be in authorized_keys:" # Show public key
-          ssh-keygen -y -f ~/.ssh/id_ed25519 # Generate public key from private
-          
-          echo "Attempting connection with verbose output:" # Test connection
-          ssh -vvv -o StrictHostKeyChecking=accept-new -o BatchMode=yes -i ~/.ssh/id_ed25519 ${{ secrets.DO_USERNAME }}@${{ secrets.DO_TAILSCALE_NAME }} 'echo "Connection successful"'
-
-      # ✅ Step 2: SSH and deploy to the droplet over private Tailscale tailnet
-      - name: SSH and Deploy with Docker Compose # Main deployment step
+          cat > envfile <<EOF  # Start heredoc to create env file
```

```suggestion
          cat > envfile << 'EOF'  # Start heredoc to create env file
```

Careful not to regress when doing making improvements on the PR.